ompi/mca/coll/base/coll_base_allreduce.c

```diff
@@ -1376,4 +1376,86 @@ int ompi_coll_base_allreduce_intra_allgather_reduce(const void *sbuf, void *rbuf
     return err;
 }
 
+int ompi_coll_base_allreduce_intra_k_bruck(const void *sbuf, void *rbuf, int count,
```

Comment:
Initially, I have another version that does reduce after each allgather (bruck) phase to save some network bandwidth because only reduced result is required to be exchanged with other ranks.  The drawback is that the reduce starts with a different element on different rank due to the wrap-around nature of bruck (potential small numerical difference) and there is very little benefit to improve allreduce latency. 

I can make either changes: 
1. change the function name to allreduce_intra_allgather;
2. or bring the added allgather module (bruck) into this function so that this function is self sufficient and doesn't depend on allgather tuning.